web/models/setting/system_variable.ex

```diff
@@ -0,0 +1,96 @@
+defmodule CoursePlanner.SystemVariable do
+  @moduledoc """
+  This module holds the model for one system variable
+  """
+  use CoursePlanner.Web, :model
+
+  alias Ecto.Changeset
+
+  schema "system_variables" do
+    field :key,      :string
+    field :value,    :string
+    field :type,     :string
+    field :visible,  :boolean
+    field :editable, :boolean
+
+    timestamps()
+  end
+
+  @doc """
+  Builds a changeset based on the `struct` and `params`.
+  """
+  def changeset(struct, params \\ %{}) do
+    target_params =
+      [
+       :key,
+       :value,
+       :type,
+       :visible,
+       :editable
+      ]
+
+    struct
+    |> cast(params, target_params)
+    |> validate_required(target_params)
+    |> validate_value_type()
+  end
+
+  def changeset(struct, params, :update) do
+    struct
+    |> changeset(params)
+    |> validate_editable()
+  end
+
+  defp validate_editable(%{valid?: true} = changeset) do
```

Comment:
for this to work properly you should not cast the value for `editable` in the update changeset, otherwise it can be bypassed by changing the editable field as well